src/lib/insights/client.ts

```diff
@@ -116,21 +116,20 @@ For more information, visit https://www.algolia.com/doc/guides/getting-insights-
  * @deprecated This function will be still supported in 4.x releases, but not further. It is replaced by the `insights` middleware. For more information, visit https://www.algolia.com/doc/guides/getting-insights-and-analytics/search-analytics/click-through-and-conversions/how-to/send-click-and-conversion-events-with-instantsearch/js/
  * It passes `insights` to `HitsWithInsightsListener` and `InfiniteHitsWithInsightsListener`.
  */
-export default function withInsights<TRendererOptions, TConnectorParams>(
-  connector: Connector<TRendererOptions, TConnectorParams>
-): Connector<TRendererOptions, TConnectorParams> {
-  const wrapRenderFn = renderFn => (renderOptions, isFirstRender) => {
-    const { results, hits, instantSearchInstance } = renderOptions;
-    if (results && hits && instantSearchInstance) {
-      const insights = wrapInsightsClient(
-        instantSearchInstance.insightsClient,
-        results,
-        hits
-      );
-      return renderFn({ ...renderOptions, insights }, isFirstRender);
-    }
-    return renderFn(renderOptions, isFirstRender);
-  };
-
-  return (renderFn, unmountFn) => connector(wrapRenderFn(renderFn), unmountFn);
+export default function withInsights<TConnector extends Connector<any, any>>(
+  connector: TConnector
+): TConnector {
+  return ((renderFn, unmountFn) =>
+    connector((renderOptions, isFirstRender) => {
+      const { results, hits, instantSearchInstance } = renderOptions;
+      if (results && hits && instantSearchInstance) {
+        const insights = wrapInsightsClient(
+          instantSearchInstance.insightsClient,
+          results,
+          hits
+        );
+        return renderFn({ ...renderOptions, insights }, isFirstRender);
+      }
+      return renderFn(renderOptions, isFirstRender);
```

Comment:
Was it not more readable isolating the renderFn wrapper, sort of to have at the end the recognisable widget wignature: `(renderFn, unmountFn) => connector(wrapRenderFn(renderFn), unmountFn);`

Reply:
actually since it's deprecated and will not be further maintained, you can ignore my comment